### PR TITLE
CORE-10076: Remove cluster level session certificate

### PR DESCRIPTION
**Pull request by yift-r3**
API pull request in https://github.com/corda/corda-api/pull/842

Remove the ability to have a cluster-level session certificate.

### CODE REVIEW

**Review by charlieR3 — DISMISSED** [review]

File: components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/KeysRestResourceImplTest.kt [hunk 1]
```diff
@@ -329,6 +331,29 @@ class KeysRestResourceImplTest {
...
 
+        @Test
+        fun `generateKeyPair throws InvalidInputDataException when the category is session init and the tenant is not a virtual node`() {
+            assertThrows<InvalidInputDataException> {
+                keysOps.generateKeyPair(tenantId = P2P, alias = ALIAS, hsmCategory = SESSION_INIT, scheme = SCHEME)
+            }
+        }
+
+        @Test
+        fun `generateKeyPair will not throw an exception when the category is session init and the tenant is not a virtual node`() {
```
Comment (williamvigorr3) [inline components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/KeysRestResourceImplTest.kt]: `the tenant is not a virtual node` -> `the tenant is a virtual node` (I think)

File: components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImpl.kt [hunk 2]
```diff
@@ -194,22 +195,61 @@ class CertificatesRestResourceImpl @Activate constructor(
...
+            throw InvalidInputDataException(
+                "No certificates in PEM"
+            )
+        }
+        if (usageType == CertificateUsage.P2P_SESSION) {
+            if (holdingIdentityShortHash == null) {
+                throw InvalidInputDataException(
+                    details = mapOf(
+                        "holdingIdentityId" to
+                            "P2P Session certificate can only be loaded to holding identity."
```
Comment (williamvigorr3) [inline components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImpl.kt]: Suggest `P2P Session certificate can only be imported for a holding identity.`

File: components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImplTest.kt [hunk 3]
```diff
@@ -583,6 +587,119 @@ class CertificatesRestResourceImplTest {
...
+                on { content } doReturn
+                    "".toByteArray().inputStream()
+            }
+            assertThrows<InvalidInputDataException> {
+                certificatesOps.importCertificateChain("rpc-api-tls", null, "alias", listOf(certificate))
+            }
+        }
+
+        @Test
+        fun `session init will fail if the certificate is a cluster certificate`() {
```
Comment (williamvigorr3) [inline components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImplTest.kt]: Suggest `session init` -> `session certificate` (and in the other test cases).

File: components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImplTest.kt [hunk 4]
```diff
@@ -583,6 +587,119 @@ class CertificatesRestResourceImplTest {
...
+                on { content } doReturn certificateText.byteInputStream()
+            }
+
+            assertThrows<InvalidInputDataException> {
+                certificatesOps.importCertificateChain("p2p-session", shortHash.value, "alias", listOf(certificate))
+            }
+        }
+
+        @Test
+        fun `session init will fail if the virtual node subject is not the member name`() {
```
Comment (williamvigorr3) [inline components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImplTest.kt]: virtual node subject -> subject

File: components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImplTest.kt [hunk 5]
```diff
@@ -583,6 +587,119 @@ class CertificatesRestResourceImplTest {
...
+                it.`when`<CertificateFactory> {
+                    CertificateFactory.getInstance("X.509")
+                }.doReturn(certificateFactory)
+
+                certificatesOps.importCertificateChain("p2p-session", shortHash.value, "alias", listOf(certificate))
+            }
+        }
+
+        @Test
+        fun `session init will fail if the certificate name is not a vaild member name`() {
```
Comment (williamvigorr3) [inline components/membership/membership-http-rpc-impl/src/test/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImplTest.kt]: Does this trigger the exception on line 236? If yes X500Principal("CN=alice, OU=MyUnit, O=MyOrg, C=GB") might be clearer than "r3.pem".

**Review by williamvigorr3 — COMMENTED** [review]
Looks good. I spotted a few minor typos.

**Review by williamvigorr3 — DISMISSED** [review]
LGTM

**Review by charlieR3 — APPROVED** [review]

File: components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImpl.kt [hunk 6]
```diff
@@ -194,22 +195,61 @@ class CertificatesRestResourceImpl @Activate constructor(
...
+            val node =
+                virtualNodeInfoReadService.getByHoldingIdentityShortHash(holdingIdentityShortHash)
+                    ?: throw InvalidInputDataException(
+                        details = mapOf(
+                            "holdingIdentityId" to
+                                "Can not find virtual node $holdingIdentityShortHash."
+                        )
+                    )
+
+            val firstCertificate = x509Certificates.first()
```
Comment (owenstanford) [inline components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImpl.kt:234]: Should we be validating that ```x509Certificates``` has a single element, what does it mean if there >1?

Comment (yift-r3) [reply components/membership/membership-http-rpc-impl/src/main/kotlin/net/corda/membership/impl/httprpc/v1/CertificatesRestResourceImpl.kt:234]: The certificate might be a certificate chain, but we are looking just at the subject of the first certificate in the chain.

**Review by owenstanford — APPROVED** [review]

**Pull request merged**